interp/interp.go

```diff
@@ -799,26 +799,46 @@ func (r *Runner) stmtSync(ctx context.Context, st *syntax.Stmt) {
 	}
 }
 
-func (r *Runner) sub() *Runner {
+// Subshell makes a copy of the given Runner, suitable for use in parallel to
+// the original.  The copy will have the same parameters, environment
+// variables, etc, but they can all be modified without affecting the
+// original.
+//
+// Subshell is not safe to run from a different goroutine if r is in the
```

Comment:
Subshell is not safe to use concurrently with Run.

Reply:
I guess it depends on how you look at it.  I was going to say exactly that, I think, but then thought about that Run can _call_ Subshell (indirectly) and I thought that was confusing: "You say I shouldn't do what you yourself are doing ... ?"  I guess that's not actually _concurrent_ use, though.  I'll change it.

Reply:
Yeah, I thought about that case too. I agree it doesn't fall under concurrent use, because Run and Subshell aren't concurrently *executing* at the same time.